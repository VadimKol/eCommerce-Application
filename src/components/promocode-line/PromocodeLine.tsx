import classNames from 'classnames';
import { toast } from 'react-toastify';

import styles from './styles.module.scss';

interface BlockInfoProps {
  colorClass: string;
  text: string;
  copyText: string;
}
export function PromocodeLine({ colorClass, text, copyText }: BlockInfoProps): JSX.Element {
  async function handleCopyClick(): Promise<void> {
    try {
      await navigator.clipboard.writeText(copyText);
      toast.success(`You copied Promo Code '${copyText}'`);
    } catch {
      toast.error(`Failed to copy Promo Code`);
    }
  }
  return (
    <div
      className={classNames(styles.animateBlock, styles[colorClass])}
      onClick={() => {
        handleCopyClick().catch(() => {});
      }}
    >
      <div className={styles.animateLine}>{text} </div>
    </div>
  );
}

export default PromocodeLine;
