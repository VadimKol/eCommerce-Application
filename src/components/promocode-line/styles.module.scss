
.animateBlock {
  position: relative;
  bottom: 0;
  left: 0;
  overflow-x: hidden;
  width: calc(100% + 40px);
  height: 40px;
  font-size: 17px;
  background-color: var(--tertiary-color);
  border-top: 3px solid var(--secondary-color);
  border-bottom: 3px solid var(--secondary-color);
}

@keyframes ticker {
  0% {
    transform: translate(0);
  }

  100% {
    transform: translate(-100%);
  }
}


@keyframes ticker2 {
  0% {
    transform: translate(100%);
  }

  100% {
    transform: translate(0);
  }
}

.animateLine {
  position: absolute;
  top: 15%;
  width: 100%;
  white-space: nowrap;
}

.animateLineFirst {
  animation: ticker 15s linear infinite;
}

.animateLineSecond {
  animation: ticker2 15s linear infinite;
}

.animateBlock:hover .animateLine {
  animation-play-state: paused;
}


@media (width <= 400px) {
  .animateLineSecond {
    display: none;
  }
}
