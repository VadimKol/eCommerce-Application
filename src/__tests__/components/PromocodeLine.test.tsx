import { render } from '@testing-library/react';

import { PromocodeLine } from '@/components/promocode-line/PromocodeLine';

it('renders correctly', () => {
  const { container } = render(<PromocodeLine colorClass="" text="" copyText="" />);
  expect(container).toMatchSnapshot();
});
