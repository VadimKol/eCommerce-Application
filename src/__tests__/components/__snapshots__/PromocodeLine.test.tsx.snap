// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="animateBlock"
  >
    <div
      class="animateLine"
    >
       
    </div>
  </div>
</div>
`;
